lms/static/js/edxnotes/utils/notes_collector.js

```diff
@@ -0,0 +1,87 @@
+(function(define) {
+    'use strict';
+    define(['jquery', 'underscore', 'annotator_1.2.9'], function($, _, Annotator) {
+        var cleanup,
+            renderNotes,
+            fetchNotesWhenReady,
+            storeNotesRequestData,
+            searchRequestsData = [];
+
+        cleanup = function () {
+            searchRequestsData = [];
+        };
+
+        storeNotesRequestData = function (data) {
+            searchRequestsData.push(data);
+            fetchNotesWhenReady();
+        };
+
+        fetchNotesWhenReady = function () {
+            if ($('.edx-notes-wrapper').length != searchRequestsData.length) {
```

Comment:
We will see what we can do. We are already too late on this so i am leaning towards merging this as currently i don't see dynamically fetching of xblocks.

@muzaffaryousaf FYI